### PR TITLE
feat: format gin logger

### PR DESCRIPTION
**Pull request by Belyenochi**
### Type of change:

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

### What this PR does / why we need it:

Resolves [#899](https://github.com/apache/apisix-ingress-controller/issues/899)

### CODE REVIEW

File: pkg/log/gin_logger.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+import (
+	"github.com/gin-gonic/gin"
+	"net"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"runtime/debug"
+	"strings"
+	"time"
```
Comment (tao12345666333) [inline pkg/log/gin_logger.go:11]: Please sort them. 
>Imports are organized in groups, with blank lines between them. The standard library packages are always in the first group.

File: pkg/log/gin_logger.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package log
```
Comment (tao12345666333) [inline pkg/log/gin_logger.go:1]: We should add Apache LICENSE header. (you can copy from other files.

**Comment by lingsamuel** [comment]
Simply run `make update-license && make update-gofmt` to update license header and imports.

File: pkg/log/gin_logger.go (hunk 2 above)
Comment (Belyenochi) [reply pkg/log/gin_logger.go:1]: > Simply run `make update-license && make update-gofmt` to update license header and imports.

Thanks!

**Comment by Belyenochi** [comment]
> Simply run `make update-license && make update-gofmt` to update license header and imports.

Thanks!

File: pkg/log/gin_logger.go (hunk 1 above)
Comment (Belyenochi) [reply pkg/log/gin_logger.go]: OK

File: pkg/log/gin_logger.go (hunk 2 above)
Comment (Belyenochi) [reply pkg/log/gin_logger.go:15]: Fixed, thanks for your review.

**Comment by lingsamuel** [comment]
Please fix this:
> Error: ../../pkg/log/gin_logger.go:26:2: missing go.sum entry for module providing package github.com/gin-gonic/gin (imported by github.com/apache/apisix-ingress-controller/pkg/log);

Thanks!

**Comment by Belyenochi** [comment]
> Please fix this:
> 
> > Error: ../../pkg/log/gin_logger.go:26:2: missing go.sum entry for module providing package github.com/gin-gonic/gin (imported by github.com/apache/apisix-ingress-controller/pkg/log);
> 
> Thanks!

Done

**Comment by Belyenochi** [comment]
@lingsamuel, hi lingsamuel, I observed ci error, check the ingress lb status is updated, I don't know what to do to fix it.

**Comment by lingsamuel** [comment]
It's a flaky test #869, just ignore it

File: pkg/log/gin_logger.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+func GinLogger(logger *Logger) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		start := time.Now()
+		path := c.Request.URL.Path
+		query := c.Request.URL.RawQuery
+		c.Next()
+		cost := time.Since(start)
+
+		logger.Infof("path: %s, status: %d, method: %s, query: %s, ip: %s, user-agent: %s, errors: %s, cost: %s",
```
Comment (lingsamuel) [inline pkg/log/gin_logger.go:37]: I am not familiar with gin-gonic, does it have similar concept like log level? wdyt @tao12345666333 ?

Comment (Belyenochi) [reply pkg/log/gin_logger.go]: @lingsamuel  @tao12345666333 I used error level alone for the return codes of 4xx and 5xx.

**Review by lingsamuel — APPROVED** [review]
LGTM

**Comment by tao12345666333** [comment]
Please check the CI errors

**Comment by Belyenochi** [comment]
> Please check the CI errors

@tao12345666333, I have some doubts about the CI error. I can't reproduce it locally. It seems to be an unstable CI unit test.

**Comment by tao12345666333** [comment]
@Belyenochi let me re-run CI.

**Comment by Belyenochi** [comment]
> @Belyenochi let me re-run CI.

It seems that everything is ok.

**Review by tao12345666333 — APPROVED** [review]

**Comment by tao12345666333** [comment]
merged!

**Pull request merged**